### PR TITLE
feat: angular 6, rxjs 6

### LINKED ISSUE #19: Angular 6 support
Hi, the angular team released an rc version. The problems that this library gives are related to the inclusions of rxjs that they updated to version 6.

They also left a migration guide to rxjs 6, [link](https://docs.google.com/document/d/12nlLt71VLKb-z3YaSGzUfx6mJbc34nsMXtByPUN35cg/preview#).

:smile:

### PR DESCRIPTION
**Pull request by osnoser1**
Maybe an alpha version can be released?

closes #19

### CODE REVIEW

**Comment by serhiisol** [comment]
Sure, I'll review pr and let you know :)

**Comment by osnoser1** [comment]
The deleted part of the documentation is because it threw errors in the compilation with messages like the following:

image

File: src/auth.interceptor.ts [hunk 1]
```diff
@@ -26,31 +19,18 @@ export class AuthInterceptor implements HttpInterceptor {
 
   /**
    * Is refresh token is being executed
-   *
```
Comment (serhiisol) [inline src/auth.interceptor.ts]: Could you put back all the docs?

File: src/auth.interceptor.ts [hunk 2]
```diff
@@ -7,16 +7,9 @@ import {
...
-import { Observable } from 'rxjs/Observable';
 
-import {
-  map,
-  first,
-  switchMap,
-  _throw,
-  _catch
-} from './rxjs.util';
+import { Observable, Subject, throwError } from 'rxjs';
```
Comment (serhiisol) [inline src/auth.interceptor.ts]: could you use explicit imports from rxjs to not to import everything from rxjs?

Comment (osnoser1) [reply src/auth.interceptor.ts]: One of the changes of rxjs 6 is that now there are no explicit imports, everything is imported from `rxjs`,` rxjs / operators`, etc. They cleaned the surface of the api in that version.

File: src/auth.interceptor.ts (hunk 1 above)
Comment (osnoser1) [reply src/auth.interceptor.ts]: Of course! :+1:

**Comment by osnoser1** [comment]
Done! The compile time error of the image that I published occurs.
With a quick search I found this [issue](https://github.com/dherges/ng-packagr/issues/485).

**Comment by serhiisol** [comment]
closed in favor of #26

**Pull request closed without merging**